core/templates/pages/exploration-editor-page/improvements-tab/services/improvement-learner-answer-details-modal.controller.spec.ts

```diff
@@ -0,0 +1,138 @@
+// Copyright 2020 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Unit tests for ImprovementLearnerAnswerDetailsModalController.
+ */
+
+describe('Improvement Learner Answer Details Modal Controller', function() {
+  var $scope = null;
+  var $uibModalInstance = null;
+  var $q;
+  var LearnerAnswerDetailsObjectFactory = null;
+  var LearnerAnswerInfoObjectFactory = null;
+  var LearnerAnswerDetailsDataService = null;
+  var learnerAnswerDetails = null;
+  var learnerAnswerInfo = null;
+
+  beforeEach(angular.mock.module('oppia'));
+  beforeEach(angular.mock.module(function($provide) {
+    $provide.value('UserExplorationPermissionsService', {
+      getPermissionsAsync: function() {
+        return $q.resolve({
+          can_edit: true
+        });
+      }
+    });
+  }));
+  beforeEach(angular.mock.inject(function($injector, $controller) {
+    LearnerAnswerDetailsObjectFactory = $injector.get(
+      'LearnerAnswerDetailsObjectFactory');
+    LearnerAnswerInfoObjectFactory = $injector.get(
+      'LearnerAnswerInfoObjectFactory');
+    LearnerAnswerDetailsDataService = $injector.get(
+      'LearnerAnswerDetailsDataService');
+    $q = $injector.get('$q');
+
+    $uibModalInstance = jasmine.createSpyObj(
+      '$uibModalInstance', ['close', 'dismiss']);
+
+    learnerAnswerInfo = LearnerAnswerInfoObjectFactory.createFromBackendDict({
+      id: '1',
+      answer: 'This is the answer',
+      answer_details: 'Details',
+      created_on: 99999
+    });
+    learnerAnswerDetails = (
+      LearnerAnswerDetailsObjectFactory.createDefaultLearnerAnswerDetails(
+        '0', 'State', '0', {
+          choices: {
+            value: 'Answer'
+          }
+        }, [learnerAnswerInfo])
+    );
+
+    var $rootScope = $injector.get('$rootScope');
+    $scope = $rootScope.$new();
+    $controller(
+      'ImprovementLearnerAnswerDetailsModalController', {
+        $scope: $scope,
+        $uibModalInstance: $uibModalInstance,
+        learnerAnswerDetails: learnerAnswerDetails
+      });
+  }));
+
+  it('should evalute scope variables value correctly', function() {
+    // $apply is called in order to resolve $q from
+    // UserExplorationPermissionsService.getPermissionsAsync.
+    $scope.$apply();
```

Comment:
Move this into the `beforeEach`, then?

Reply:
I've removed all the async calls from the controllers, so `$apply` is not necessary anymore.